osu.Game/Overlays/BeatmapSet/BeatmapRulesetSelector.cs

```diff
@@ -11,6 +11,8 @@ namespace osu.Game.Overlays.BeatmapSet
 {
     public class BeatmapRulesetSelector : OverlayRulesetSelector
     {
+        protected override bool SelectInitialRuleset => false;
```

Comment:
I'm not sure I understand why this is required. Wouldn't it make sense for all overlays to follow this pattern?

Reply:
Not in this case, no. The beatmap set overlay has a custom implementation of ruleset selecting that involves disabling rulesets which the beatmapset doesn't have beatmaps for.

In my upcoming refactor of this beatmap set overlay to follow the new designs, I have rewrote it entirely to take the rulesets from the beatmap rather than adding all the available rulesets and disabling ones which don't have any beatmaps, rather than this workaround.

Just to be clear, I'm not a fan of this at all, but I've done it as a workaround for the time being, I'll add a comment to be completely explicit about that.

Reply:
> Not in this case, no. The beatmap set overlay has a custom implementation of ruleset selecting that involves disabling rulesets which the beatmapset doesn't have beatmaps for.

Do you mean.. the beatmap info overlay?

Can't you just have it overwrite the "initial" ruleset by running its value change after the `base` one?

Reply:
Yeah beatmap info overlay. Sure can overwrite it instead.